src/Stratis.SmartContracts.Core.Validation/Validators/Method/MultiDimensionalArrayValidator.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Stratis.ModuleValidation.Net;
+
+namespace Stratis.SmartContracts.Core.Validation
+{
+    /// <summary>
+    /// Validates that methods don't contain any multi-dimensional array initializers.
+    /// </summary>
+    public class MultiDimensionalArrayValidator : IMethodDefinitionValidator
+    {
+        public static readonly string ErrorType = "Multi-Dimensional Arrays Not Supported";
+
+        public IEnumerable<ValidationResult> Validate(MethodDefinition method)
+        {
+            if (!method.HasBody || method.Body.Instructions.Count == 0)
+                return Enumerable.Empty<MethodDefinitionValidationResult>();
+
+            foreach (Instruction instruction in method.Body.Instructions)
+            {
+                if (instruction.OpCode.Code == Code.Newobj)
```

Comment:
There's an `IInstructionValidator` for these circumstances. Prevents us from unnecessarily looping over instructions in a method body multiple times. See `TypePolicyValidator.cs` for how it gets used.